### PR TITLE
PaPaYoU - Introducing LUA and OSD Updates to utilize the new Filter parameters

### PR DESCRIPTION
**Pull request by papayou66**
Lua stuff
add imuf

### CODE REVIEW

**Comment by kidBrazil** [comment]
hey @papayou66 what is the status of your testing with this so far?

**Comment by kidBrazil** [comment]
@papayou66 can you please take a look at Travis - seems there is an issue https://travis-ci.org/ButterFlight/butterflight/jobs/354760664

File: src/main/common/filter.h [hunk 1]
```diff
@@ -70,9 +70,9 @@ typedef enum {
 } filterType_e;
 
 typedef enum {
-    KD_FILTER_CLASSIC = 0,
```
Comment (kidBrazil) [inline src/main/common/filter.h]: Hate to do this to ya PaPaYou but I think we should revert this change and keep them as KD_FILTER_XXXX - just so the code clarity is improved.

SP / CLASSIC / NOSP are fine for people familiar with the genesis of these features but for anyone else that comes in on this code it might be confusing. The KD_FILTER add some clarity by making it immeditaly obvious that its dealing with D term filters.

File: src/main/flight/pid.c [hunk 2]
```diff
@@ -108,7 +108,7 @@ void resetPidProfile(pidProfile_t *pidProfile)
         .dterm_notch_hz = 260,
         .dterm_notch_cutoff = 160,
         .dterm_filter_type = FILTER_PT1,
-        .dterm_filter_style = KD_FILTER_CLASSIC,
+        .dterm_filter_style = CLASSIC,
```
Comment (kidBrazil) [inline src/main/flight/pid.c]: Revert to KD_FILTER_CLASSIC

File: src/main/flight/pid.c [hunk 3]
```diff
@@ -511,28 +511,28 @@ void pidController(const pidProfile_t *pidProfile, const rollAndPitchTrims_t *an
...
             const float rD = dynCd * MIN(getRcDeflectionAbs(axis) * relaxFactor, 1.0f) * currentPidSetpoint - gyroRateFiltered;    // cr - y
             const float pureRD = dynCd * MIN(getRcDeflectionAbs(axis) * relaxFactor, 1.0f) * getSetpointRate(axis) - gyroRateFiltered;    // cr - y
-            
+
             float delta = 0.0f;
             float iDT = 1.0f/deltaT; //divide once
 
             switch (pidProfile->dterm_filter_style) {
-                case KD_FILTER_CLASSIC:
+                case CLASSIC:
```
Comment (kidBrazil) [inline src/main/flight/pid.c]: Revert cases to KD_FILTER_XXXX variant

File: src/main/interface/msp.c [hunk 4]
```diff
@@ -1178,16 +1178,32 @@ static bool mspProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst)
...
         break;
+
 #ifdef USE_GYRO_FAST_KALMAN
-    case MSP_ADVANCED_FILTER_CONFIG :
-        sbufWriteU16(dst, gyroConfig()->gyro_soft_lpf_hz_2);
+    case MSP_FAST_KALMAN :
         sbufWriteU16(dst, gyroConfig()->gyro_filter_q);
         sbufWriteU16(dst, gyroConfig()->gyro_filter_r);
-        sbufWriteU16(dst, gyroConfig()->gyro_filter_p);
-        sbufWriteU8(dst, gyroConfig()->gyro_stage2_filter_type);
```
Comment (kidBrazil) [inline src/main/interface/msp.c]: Do we not need gyro_stage2_filter_type anymore?

File: src/main/interface/msp.c [hunk 5]
```diff
@@ -1178,16 +1178,32 @@ static bool mspProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst)
...
         break;
 #endif
+
+#ifdef USE_GYRO_IMUF9001
```
Comment (kidBrazil) [inline src/main/interface/msp.c]: Something whack going on with the TABS on this area - could we please get that harmonized to line up with the rest?

File: src/main/interface/settings.c [hunk 6]
```diff
@@ -250,7 +250,7 @@ static const char * const lookupTableLowpassType[] = {
 };
 
 static const char * const lookupTableKdStyle[] = {
-    "KD_FILTER_CLASSIC", "KD_FILTER_SP", "KD_FILTER_NOSP"
```
Comment (kidBrazil) [inline src/main/interface/settings.c]: Please revert for clarity =D

**Comment by papayou66** [comment]
Travis is my friend today

File: src/main/target/HELIOSPRING/config.c [hunk 7]
```diff
@@ -80,12 +80,11 @@ void targetConfiguration(void) {
         pidProfile->itermAcceleratorGain = 5000;
         // should't need to set these since they don't get init in gyro.c with USE_GYRO_IMUF
```
Comment (kidBrazil) [inline src/main/target/HELIOSPRING/config.c]: If these are really no longer needed we should just remove them entirely instead of leaving comments IMO

File: src/main/common/filter.h (hunk 1 above)
Comment (papayou66) [reply src/main/common/filter.h]: Yes sure no problem

File: src/main/interface/msp.c (hunk 5 above)
Comment (papayou66) [reply src/main/interface/msp.c]: done

**Comment by papayou66** [comment]
Switch to a new branch to avoid merge conflict
